kbfsgit/runner.go

```diff
@@ -1501,17 +1503,49 @@ func (r *runner) pushSome(
 			err = nil
 		}
 
+		// TODO (KBFS-2682): also load parent refs.
+		refs := make([]*plumbing.Reference, 0, len(refspecs))
+
 		// All non-delete refspecs in the batch get the same error.
 		for _, refspec := range refspecs {
 			refStr := refspec.String()
 			start := strings.Index(refStr, ":") + 1
 			dst := refStr[start:]
 			results[dst] = err
+			ref, refErr := repo.Storer.Reference(plumbing.ReferenceName(dst))
+			if refErr != nil {
+				r.log.CDebugf(ctx, "Error loading ref for refspec %s: %+v",
+					refspec, refErr)
+				continue
+			}
+			refs = append(refs, ref)
 		}
 	}
 	return results, nil
 }
 
+func (r *runner) refsToCommits(ctx context.Context,
+	storer gogitstor.EncodedObjectStorer,
+	refs []*plumbing.Reference) ([]*gogitobj.Commit, error) {
+	erroredRefs := make(map[string]error, 0, len(refs))
+	commits := make([]*gogitobj.Commit, 0, len(refs))
+	for _, ref := range refs {
+		commit, err := gogitobj.GetCommit(storer, ref.Hash())
```

Comment:
To get the parent commits here, you can use `gogit.Repository.Log()` for the local `Repository` you need to create, and pass it the commit hash.  You'll get back an iterator that you can follow as long as desired.